src/components/Config.tsx

```diff
@@ -0,0 +1,170 @@
+        history.pushState(null, null, event.currentTarget.hash);
+
+        const fullURL = window.location.href.split("#")[0];
+        const hash = createUrlHash(parentKey, name);
+        navigator.clipboard.writeText(fullURL + "#" + hash);
+        scrollIntoView(hash);
+
+        event.stopPropagation();
+    };
+
+    const showAndScrollIntoView = hash => {
+        setShowDescription(true);
+        scrollIntoView(hash);
+    }
+
+    return (
+        <div
+            style={{ paddingLeft: `${INDENT_SIZE}px` }}
+            id={createUrlHash(parentKey, name)}
+        >
+            <div className={`description_word_wrap`}
+                 style={{ marginBottom: showDescription ? 10 : 0 }}
+            >
+                <a
+                    onClick={() => {setShowDescription(!showDescription)}}
+                    className={`with-value${showDescription ? "-active": ""}`}
+                >
+                    {name}: {node.default.toString()}
+                </a>
+                {showDescription ? (
+                    <>
+                        <a
+                            className={`config-anchor with-value-active-colour hash-link`}
+                            href={`#${createUrlHash(parentKey, name)}`}
+                            onClick={handleHashLinkClick}
+                        ></a>
+
+                        <div className="indent-2" style={{ marginBottom: 10 }}>
+                            <div className="outlined-box description-text colour-offset-box">
+                                <ReactMarkdown className={style.reactMarkDown}>{node.description.toString()}</ReactMarkdown>
+                            </div>
+                        </div>
+                    </>
+                ) : (
+                    <a
+                        className={`config-anchor with-value-active-colour hash-link`}
+                        href={`#${createUrlHash(parentKey, name)}`}
+                        onClick={handleHashLinkClick}
+                    ></a>
+                )}
+            </div>
+        </div>
+    );
+};
+
+const renderYamlData = (
+    data: YamlData,
+    parentKey?: string,
+    root: boolean = false
+): JSX.Element[] => {
+    const renderedNodes: JSX.Element[] = [];
+
+    for (const [key, value] of Object.entries(data)) {
+        if (typeof value === 'object') {
+            if ('default' in value || 'description' in value) {
+                renderedNodes.push(
+                    <YamlNodeWithDescription
+                        key={key}
+                        name={key}
+                        parentKey={parentKey}
+                        node={value as YamlNode}
+                    />
+                );
+            } else {
+                renderedNodes.push(
+                    <div
+                        key={key}
+                        style={{ paddingLeft: `${root ? 0 : INDENT_SIZE}px` }}
+                    >
```

Comment:
I don't know how it would look, but putting a left border on this div would create those vertical lines between sections which can be helpful in trying to figure out what section something is in. Then you could have the border "light up" when the div is being hovered over as further indication of what section you are in.